packages/visual-dev/src/components/Buttons.tsx

```diff
@@ -0,0 +1,352 @@
+  }
+`;
+
+export const ButtonSpinner = () => {
+  return (
+    <ButtonSpinnerStyle>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+    </ButtonSpinnerStyle>
+  );
+};
+
+export const PrimaryButton = ({
+  children,
+  onClick,
+  padding,
+  buttonColor,
+  darkercolor,
+  disabledcolor,
+  loading,
+  ...props
+}: PBProps) => {
+  return (
+    <PrimaryButtonStyled
+      {...props}
+      padding={padding}
+      buttonColor={buttonColor}
+      darkercolor={darkercolor}
+      disabledcolor={disabledcolor}
+      onClick={onClick}
+      disabled={loading || props.disabled}
+    >
+      {children}
+      {loading && <ButtonSpinner />}
+    </PrimaryButtonStyled>
+  );
+};
+
+// TODO: make one Primary Button to rule them all!
+const PrimaryButtonStyled = styled.button<PBProps>`
+  border-radius: 20px;
+  min-width: 100px;
+  padding: ${(props) => props.padding || `3px 19px`};
+  background: ${(props) => props.buttonColor || `#f5a841`};
+  border: 1px solid ${(props) => props.buttonColor || `#f5a841`};
+  color: #fff;
+  font-weight: 600;
+  font-size: 13px;
+  font-family: "Helvetica Neue", Helvetica, sans-serif;
+  &:hover,
+  &:active,
+  &:focus {
+    background: ${(props) => props.darkercolor || `#DC8F28`};
+    border: 1px solid ${(props) => props.buttonColor || `#DC8F28`};
+    color: #fff;
+    text-decoration: none;
+    outline: none;
+  }
+  :disabled {
+    background: ${(props) => props.disabledcolor || `#E2E2E2`};
+    border: 1px solid ${(props) => props.disabledcolor || `#E2E2E2`};
+    pointer-events: none;
+  }
+  ${ButtonDiv} & {
+    margin: 0 8px;
+  }
+`;
+
+
+
+// Primary Button ends
+
+interface IPrimaryButton extends PBProps {
+  onClick: any;
+  disabled?: boolean;
+  success?: boolean;
+  successText?: string;
+  children: string;
+  loading: boolean;
+}
+
+export const PrimaryButtonNew = (props: IPrimaryButton) => {
+  const [justClicked, setJustClicked] = useState(false);
+
+  const {
+    onClick,
+    disabled,
+    success,
+    buttonColor,
+    darkercolor,
+    disabledcolor,
+    padding,
+    successText,
+    children,
+    loading,
+  } = props;
+
+  const successfulClick = success && justClicked;
+
+  // TODO: how do I check if the onClick is successful? (I guess the success result comes from the props.success)
+  // TODO: build in a way to have the checkmark persist instead of going away.
+  async function onClickTimer(e : SyntheticEvent) {
+    try {
+      await onClick(e);
+      setJustClicked(true);
+      await new Promise((resolve) => setTimeout(resolve, 5000));
+      setJustClicked(false);
+    } catch (e) {
+      throw new Error(e);
+    }
+  }
+
+  return (
+    <PrimaryButton
+      buttonColor={buttonColor || (successfulClick && `#57AC59`)}
+      darkercolor={darkercolor || (successfulClick && `#57AC59`)}
+      disabledcolor={disabledcolor}
+      padding={padding || (successfulClick && `3px 30px 3px 19px`)}
```

Comment:
If you specify a value for the `padding` prop, then the overrided padding on `successfulClick` will not be applied - is this intended? If you are meaning to override the padding when `successfulClick` is true in all cases then it should be:
```suggestion
      padding={successfulClick ? `3px 30px 3px 19px` : padding}
```